superduperdb/db/base/metadata.py

```diff
@@ -135,59 +126,213 @@ def get_component(
+        """
+        Hide a component version.
+
+        :param type_id: Type of component
+        :param identifier: Identifier of component
+        :param version: Version of component
+        """
+        pass
+
+    @abstractmethod
+    def _replace_component(self, info, identifier, type_id, version):
+        pass
+
+    def replace_component(
         self,
+        info: t.Dict[str, t.Any],
         identifier: str,
         type_id: str,
-        key: str,
-        value: t.Any,
-        version: int,
-    ):
-        pass
+        version: t.Optional[int] = None,
+    ) -> None:
+        """
+        Replace a component in the metadata store.
 
-    def update_object(self, identifier, type_id, key, value, version=None):
+        :param info: Information about the component found in ``component.serialize()``
+        :param identifier: Identifier of component
+        :param type_id: Type of component
+        :param version: Version of component
+        """
         if version is not None:
             version = self.get_latest_version(type_id, identifier)
-        return self._update_object(
+        return self._replace_component(
+            info=info,
             identifier=identifier,
             type_id=type_id,
-            key=key,
-            value=value,
             version=version,
         )
 
     @abstractmethod
-    def _replace_object(self, info, identifier, type_id, version):
+    def show_components(self, type_id: str, **kwargs):
+        """
+        Show components in the metadata store.
+
+        :param type_id: Type of component
+        """
+        pass
+
+    @abstractmethod
+    def show_component_versions(self, type_id: str, identifier: str):
+        """
+        Show component versions in the metadata store.
+
+        :param type_id: Type of component
+        :param identifier: Identifier of component
+        """
         pass
 
-    def replace_object(
+    @abstractmethod
+    def _update_component(
         self,
-        info: t.Dict[str, t.Any],
         identifier: str,
         type_id: str,
-        version: t.Optional[int] = None,
-    ) -> None:
+        key: str,
+        value: t.Any,
+        version: int,
+    ):
+        pass
+
+    def update_component(self, identifier, type_id, key, value, version=None):
+        """
+        Update a component in the metadata store.
+
+        :param identifier: Identifier of component
+        :param type_id: Type of component
+        :param version: Version of component (optional - defaults to latest)
+        """
         if version is not None:
             version = self.get_latest_version(type_id, identifier)
-        return self._replace_object(
-            info=info,
+        return self._update_component(
             identifier=identifier,
             type_id=type_id,
+            key=key,
+            value=value,
             version=version,
         )
 
+    # --------------- JOBS ------------------
```

Comment:
In SQLite, it probably won't make sense to log jobs to the DB, since concurrent writes aren't possible. For that reason we should separate logging from the meta-data store.